src/cli/serve/serve.js

```diff
@@ -322,18 +312,16 @@ export default function (program) {
     const configs = [getConfigPath(), ...getEnvConfigs(), ...(opts.config || [])];
     const serverlessMode = getServerlessProjectMode(opts);
 
-    // we "unshift" .serverless. config so that it only overrides defaults
     if (serverlessMode) {
-      maybeAddConfig(`serverless.yml`, configs, 'push');
-      maybeSetRecentConfig('serverless.recent.yml', serverlessMode, configs, 'unshift');
+      maybeSetRecentConfig('serverless.recent.yml', serverlessMode, configs, 'push');
     }
 
     // .dev. configs are "pushed" so that they override all other config files
     if (opts.dev && opts.devConfig !== false) {
       maybeAddConfig('kibana.dev.yml', configs, 'push');
       if (serverlessMode) {
         maybeAddConfig(`serverless.dev.yml`, configs, 'push');
-        maybeSetRecentConfig('serverless.recent.dev.yml', serverlessMode, configs, 'unshift');
+        maybeAddConfig('serverless.recent.dev.yml', configs, 'push');
```

Comment:
Using `maybeAddConfig` because otherwise `serverless.recent.dev.yml` and `serverless.recent.yml` will be the exact copies. IIUC, this file is used to manually override something in dev mode.

Reply:
Awesome catch.